Source/Dafny/AST/DafnyAst.cs

```diff
@@ -24,32 +24,56 @@ public class FilledInDuringResolutionAttribute : System.Attribute { }
   public interface IToken : Microsoft.Boogie.IToken {
     // Inherited for now.
     /*
-    int kind { get; set; }
-    string filename { get; set; }
-    int pos { get; set; }
-    int col { get; set; }
-    int line { get; set; }
-    string val { get; set; }
+    int kind { get; set; } // Used by coco, so we can't rename it to Kind
+    string filename { get; set; } // Could be renamed to Filename, for another time.
+    int pos { get; set; } // Used by coco, so we can't rename it to Pos
+    int col { get; set; } // Used by coco, so we can't rename it to Col
+    int line { get; set; } // Used by coco, so we can't rename it to Line
+    string val { get; set; } // Used by coco, so we can't rename it to Val
     bool IsValid { get; }*/
-    string leadingTrivia { get; set; }
-    string trailingTrivia { get; set; }
+    string Boogie.IToken.filename {
+      get => Filename;
+      set => Filename = value;
+    }
+
+    string Filename { get; set; }
+
+    /// <summary>
+    /// TrailingTrivia contains everything after the token,
+    /// until and including two newlines between which there is no commment
+    /// All the remaining trivia is for the LeadingTrivia of the next token
+    ///
+    /// ```
+    /// const /*for const*/ x /*for x*/ := /* for := */ 1/* for 1 */
+    /// // for 1 again
```

Comment:
How can `// for 1 again` become a docstring for the const, if the trivia is attached to a childnode of the body of the const?